src/dune_file.ml

```diff
@@ -160,12 +163,12 @@ module Pps_and_flags = struct
       let+ l, flags =
         until_keyword "--"
           ~before:(plain_string (fun ~loc s -> (loc, s)))
-          ~after:(repeat string)
+          ~after:(repeat String_with_vars.decode)
```

Comment:
We need to error out if the user put a variable here and is using a version of the dune language that is older that 1.10. We do this in `Rule.Targets.decode_static` in the same file.